### PR TITLE
CI:  update Arcade to fix CI error 

### PR DESCRIPTION
**Pull request by dtivel**
Resolve https://github.com/dotnet/sign/issues/689.

This change copies latest changes from https://github.com/dotnet/arcade/tree/release/8.0.

CC @clairernovotny, @erdembayar

### CODE REVIEW

File: eng/common/templates-official/job/source-index-stage1.yml [hunk 1]
```diff
@@ -62,7 +62,24 @@ jobs:
...
+    - task: AzureCLI@2
+      displayName: Get stage 1 auth token
+      inputs:
+        azureSubscription: 'SourceDotNet Stage1 Publish'
+        addSpnToEnvironment: true
+        scriptType: 'ps'
+        scriptLocation: 'inlineScript'
+        inlineScript: |
+          echo "##vso[task.setvariable variable=ARM_CLIENT_ID]$env:servicePrincipalId"
+          echo "##vso[task.setvariable variable=ARM_ID_TOKEN]$env:idToken"
```
Comment (erdembayar) [inline eng/common/templates-official/job/source-index-stage1.yml:74]: Not sure for Github, but Azure has this `;issecret=true` https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-secret-variables?view=azure-devops&tabs=yaml%2Cbash#set-secret-variable-in-a-script-with-logging-commands

Does it apply here?

`echo "##vso[task.setvariable variable=ARM_ID_TOKEN;issecret=true]$env:idToken"`

File: eng/common/templates-official/job/publish-build-assets.yml [hunk 2]
```diff
@@ -60,8 +60,8 @@ jobs:
       os: windows
     # If it's not devdiv, it's dnceng
     ${{ if ne(variables['System.TeamProject'], 'DevDiv') }}:
-      name: $(DncEngInternalBuildPool)
-      image: 1es-windows-2022-pt
+      name: NetCore1ESPool-Publishing-Internal
+      image: windows.vs2019.amd64
```
Comment (erdembayar) [inline eng/common/templates-official/job/publish-build-assets.yml:64]: Don't they have 2022 image?

File: eng/common/templates-official/job/source-index-stage1.yml (hunk 1 above)
Comment (dtivel) [reply eng/common/templates-official/job/source-index-stage1.yml:74]: It's a federated token, which isn't a secret in the traditional sense.  The whole idea behind federated credentials is to eliminate secret management.

Comment (dtivel) [reply eng/common/templates-official/job/source-index-stage1.yml:74]: See https://github.com/dotnet/arcade/pull/14750/files#r1600647304

File: eng/common/templates-official/job/publish-build-assets.yml (hunk 2 above)
Comment (dtivel) [reply eng/common/templates-official/job/publish-build-assets.yml:64]: The source for this change is https://github.com/dotnet/arcade/pull/14663.

Is this important?  As long as CI builds keep working, it seems OK.  It has nothing to do with the .NET SDK version we use to build Sign CLI.  The Arcade team changed agent pools.  Maybe they're planning to update images within the pool.  A benefit of Arcade is that the Arcade team manages these details, so we don't have to.

**Review by erdembayar — APPROVED** [review]

**Pull request merged**